diff --git a/dist/rollup-plugin-typescript2.cjs.js b/dist/rollup-plugin-typescript2.cjs.js
index 9ab972b041cc76f8f786d6a20f3efb53c364cad6..13e056a3c0971eb18b307d91fad096a9f3b9de79 100644
--- a/dist/rollup-plugin-typescript2.cjs.js
+++ b/dist/rollup-plugin-typescript2.cjs.js
@@ -29799,6 +29799,13 @@ const typescript = (options) => {
                 declarations[key] = { type: result.dts, map: result.dtsmap };
                 context.debug(() => `${safe.exports.blue("generated declarations")} for '${key}'`);
             }
+            // if a user sets this compilerOption, they probably want another plugin (e.g. Babel, ESBuild) to transform their TS instead, while rpt2 just type-checks and/or outputs declarations
+            // note that result.code is non-existent if emitDeclarationOnly per https://github.com/ezolenko/rollup-plugin-typescript2/issues/268
+            if (parsedConfig.options.emitDeclarationOnly)
+            {
+                context.debug(() => `${blue("emitDeclarationOnly")} enabled, not transforming TS'`);
+                return undefined;
+            }
             const transformResult = { code: result.code, map: { mappings: "" } };
             if (result.map) {
                 if (pluginOptions.sourceMapCallback)
diff --git a/dist/rollup-plugin-typescript2.es.js b/dist/rollup-plugin-typescript2.es.js
index e43bf8f03bc6792b61d8352e04bb6466712426c2..420e8f0d0d109076bc72e9d60240077235a9ba11 100644
--- a/dist/rollup-plugin-typescript2.es.js
+++ b/dist/rollup-plugin-typescript2.es.js
@@ -29770,6 +29770,13 @@ const typescript = (options) => {
                 declarations[key] = { type: result.dts, map: result.dtsmap };
                 context.debug(() => `${safe.exports.blue("generated declarations")} for '${key}'`);
             }
+            // if a user sets this compilerOption, they probably want another plugin (e.g. Babel, ESBuild) to transform their TS instead, while rpt2 just type-checks and/or outputs declarations
+            // note that result.code is non-existent if emitDeclarationOnly per https://github.com/ezolenko/rollup-plugin-typescript2/issues/268
+            if (parsedConfig.options.emitDeclarationOnly)
+            {
+                context.debug(() => `${blue("emitDeclarationOnly")} enabled, not transforming TS'`);
+                return undefined;
+            }
             const transformResult = { code: result.code, map: { mappings: "" } };
             if (result.map) {
                 if (pluginOptions.sourceMapCallback)
