diff --git a/dist/cjs/typescript.cjs b/dist/cjs/typescript.cjs
index 92f36188e187f0b80b4ea633b40cf530edadaa22..bb8ec769a6aa047f90cb3b435ff3c08924d4875b 100644
--- a/dist/cjs/typescript.cjs
+++ b/dist/cjs/typescript.cjs
@@ -344,7 +344,7 @@ var rules = {
   "sonarjs/no-identical-conditions": 2,
   "sonarjs/no-identical-expressions": 2,
   "sonarjs/no-ignored-return": 2,
-  "sonarjs/no-one-iteration-loop": 2,
+  // "sonarjs/no-one-iteration-loop": 2,
   "sonarjs/no-use-of-empty-return-value": 2,
   "sonarjs/non-existent-operator": 2,
   "sonarjs/elseif-without-else": 0,
diff --git a/dist/esm/typescript.mjs b/dist/esm/typescript.mjs
index 40a872af5d51f37dcdcc8f9098568d176ca0f122..21c4426fefc405536027d3fcb58261613d0128e3 100644
--- a/dist/esm/typescript.mjs
+++ b/dist/esm/typescript.mjs
@@ -335,7 +335,7 @@ var rules = {
   "sonarjs/no-identical-conditions": 2,
   "sonarjs/no-identical-expressions": 2,
   "sonarjs/no-ignored-return": 2,
-  "sonarjs/no-one-iteration-loop": 2,
+  // "sonarjs/no-one-iteration-loop": 2,
   "sonarjs/no-use-of-empty-return-value": 2,
   "sonarjs/non-existent-operator": 2,
   "sonarjs/elseif-without-else": 0,
